src/x86/sse.asm

```diff
@@ -0,0 +1,510 @@
+; Copyright (c) 2020, The rav1e contributors. All rights reserved
+;
+; This source code is subject to the terms of the BSD 2 Clause License and
+; the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+; was not distributed with this source code in the LICENSE file, you can
+; obtain it at www.aomedia.org/license/software. If the Alliance for Open
+; Media Patent License 1.0 was not distributed with this source code in the
+; PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+%include "ext/x86/x86inc.asm"
+
+%if ARCH_X86_64
```

Comment:
Don't think this is needed, most of the functions use less than 8 registers.
You can have a separate WEIGHTED_SSE X, Y if block for those that go above 8.
But no strong opinion on this if it makes the rust code more complicated.

Reply:
I don't think much assembly is hooked up for x86-32, but I'll gate that won't work on x86 for when we get around to it.